### PR TITLE
feat: demo Composition API

### PR DESCRIPTION
**Pull request by 0song**
Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: website/components/demo-block.vue [hunk 1]
```diff
@@ -28,9 +28,20 @@
...
       </transition>
+      <div class="control-button-container control-button-container-left">
+        <el-button
+          v-show="isExpanded && hasSetup"
+          size="small"
+          type="text"
+          class="control-button"
+          @click.stop="onSwitchSyntax"
+        >
+          {{ showSetup? langConfig['switch-button-option-text'] : langConfig['switch-button-setup-text'] }}
```
Comment (JeremyWuuuuu) [inline website/components/demo-block.vue:39]: Add a space after `showSetup`

File: website/components/demo-block.vue [hunk 2]
```diff
@@ -61,15 +72,20 @@ import { nextTick } from 'vue'
...
 const stripTemplateAndRemoveTemplate = code => {
-  const result = stripTemplate(code)
+  const result = removeSetup(stripTemplate(code))
   if (result.indexOf('<template>') === 0) {
     return result.replace(/^<template>/, '').replace(/<\/template>$/,'')
   }
   return result
 }
+const sanitizeHTML = str => {
+  var temp = document.createElement('div')
```
Comment (JeremyWuuuuu) [inline website/components/demo-block.vue:85]: Maybe using `createDocumentFragment` API will be better? This is not a mandatory change though.

File: website/components/demo-block.vue [hunk 3]
```diff
@@ -151,31 +165,62 @@ export default {
...
+    prettyCode () {
+      nextTick(() => {
+        const highlight = this.$el.querySelector('.highlight')
+        const hlcode = highlight.querySelector('pre code')
+        hlcode.innerHTML = sanitizeHTML(`<template>${this.codepen.html}</template>
+
+<script>
+  ${this.displayDemoCode}
+${'</sc' + 'ript>'}
+`)
```
Comment (JeremyWuuuuu) [inline website/components/demo-block.vue:204]: Should we format this code?

**Review by jw-foss — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
cc @XingXiaoWu, Hi, I'd like you to see the example of how to add `Composition API` demo into ElementPlus.

File: website/components/demo-block.vue (hunk 1 above)
Comment (0song) [reply website/components/demo-block.vue:39]: 1

**Comment by XingXiaoWu** [comment]
> cc @XingXiaoWu, Hi, I'd like you to see the example of how to add `Composition API` demo into ElementPlus.

It looks like add a button to change `Options API` or `Composition API`, you means I can add `Composition API` in this new way?

**Comment by 0song** [comment]
@XingXiaoWu You can check the first demo here https://preview-2099-element-plus.surge.sh/#/zh-CN/component/radio

We have added a new feature to switch demo code between Composition API / Options API 

image

**Comment by XingXiaoWu** [comment]
> @XingXiaoWu You can check the first demo here https://preview-2099-element-plus.surge.sh/#/zh-CN/component/radio
> 
> We have added a new feature to switch demo code between Composition API / Options API
> 
> image
> 
> And all we have to do is to add a ` block in the demo code like this
> 
> ```
> 
> ```
> 
> image

Wow! It's really a good idea! I wiil change the code for `Composition API` in free time

**Comment by JeremyWuuuuu** [comment]
@XingXiaoWu Thanks! Shout out for @0song for making such a good contribution!

**Pull request merged**